tuf/ngclient/updater.py

```diff
@@ -475,3 +570,61 @@ def _preorder_depth_first_walk(
 def _ensure_trailing_slash(url: str) -> str:
     """Return url guaranteed to end in a slash"""
     return url if url.endswith("/") else f"{url}/"
+
+
+def _get_spec_version(
+    repository_versions: List[str],
+    spec_version: str,
+    supported_versions: List[str],
+) -> Tuple[str, Optional[str]]:
```

Comment:
Might be nice to introduce a `SpecVersion` class (basically a `version`/`path` tuple)

Reply:
This would make the `int` conversion lines a little more bulletproof